knora/dsplib/utils/xml_upload.py

```diff
@@ -26,6 +28,8 @@
 from knora.dsplib.models.xmlresource import XMLResource
 from knora.dsplib.utils.shared import try_network_action, validate_xml_against_schema
 
+MetricRecord = namedtuple("MetricRecord", ["res_id", "filetype", "filesize_mb", "event", "duration_microsec", "mb_per_sec"])
```

Comment:
I like this pattern of using a data structure ("product type" in functional terminology) as a model for data in code. Personally I tend to implement this using dataclasses rather than namedtuples... but that's just a matter of taste.